mypy/semanal.py

```diff
@@ -1822,6 +1822,15 @@ def visit_import_from(self, imp: ImportFrom) -> None:
             module_public = use_implicit_reexport or (as_id is not None and id == as_id)
 
             if node and not node.module_hidden:
+                # expand into importing class, too... unless there's something already there.
+                # (deciding what expands is based on line order which cannot be checked here.)
+                if self.is_class_scope():
+                    assert self.type is not None
+                    defn = node.node
+                    if imported_id not in self.type.names.keys() and isinstance(defn, FuncDef):
+                        if not defn.is_decorated and not defn.is_overload:
+                            defn.info = self.type
+                            self.add_symbol(defn.name, defn, imp)
```

Comment:
Unfortunately this doesn't seem right/enough. We modify the function as defined in the original module, but it should only be treated as a method in this class, not elsewhere. I wonder what would happen if import the same function into multiple classes. For completeness, we should probably do this for `Decorator` and `OverloadedFuncDef` as well. 

Implementing this in a full and correct way seems somewhat complicated. We could perhaps take a copy of the original `FuncDef` (or other node) and only modify the copy. 

Alternatively, you could detect this and generate a blocking error, and not add anything to the symbol table. This would be the easiest fix. The error message can contain a note that suggests doing something like `name = mod.func` instead.